project-1/albertafriadii/pkg/books/delivery/http/book.go

```diff
@@ -0,0 +1,281 @@
+package http
+
+import (
+	"errors"
+	"golang-mentoring/project-1/albertafriadii/pkg/config"
+	"golang-mentoring/project-1/albertafriadii/pkg/domain"
+	"net/http"
+	"strconv"
+
+	"github.com/labstack/echo/v4"
+)
+
+type BookController struct {
+	usecase domain.BookUsecaseInterface
+}
+
+func NewBookController(bu domain.BookUsecaseInterface) *BookController {
+	return &BookController{
+		usecase: bu,
+	}
+}
+
+func (c *BookController) ListBook(ctx echo.Context) error {
+	book, err := c.usecase.ListBook(ctx.Request().Context())
+	if err != nil {
+		if errors.Is(err, config.ErrNotFound) {
+			return domain.ErrResponse(ctx, http.StatusNotFound, config.ErrNotFound)
+		}
+
+		return domain.ErrResponse(ctx, http.StatusInternalServerError, config.ErrInternalServerError)
+	}
+
+	return domain.SuccessDataResponse(ctx, book)
+}
+
+func (c *BookController) CreateBook(ctx echo.Context) error {
+	var b domain.InputCreateBook
+	var err error
+
+	bindErr := ctx.Bind(&b)
+	if bindErr != nil {
+		return domain.ErrResponse(ctx, http.StatusBadRequest, errors.New("data bind error"))
+	}
+
+	validationErr := config.Validator(&b)
+	if validationErr != nil {
+		return domain.ErrValidResponse(ctx, http.StatusBadRequest, validationErr)
+	}
+
+	if !config.ValidationBook(b.Author) {
+		return domain.ErrResponse(ctx, http.StatusBadRequest, errors.New("Pengarang Must Uppercase"))
+	}
+
+	_, err = c.usecase.CreateBook(ctx.Request().Context(), b.ToCreateBookDomain())
+	if err != nil {
+		errStatus, errMessage := config.ErrInputBookCheck(err)
+		return domain.ErrResponse(ctx, errStatus, errMessage)
+	}
+	return domain.OkResponse(ctx)
+}
+
+func (c *BookController) UpdateBook(ctx echo.Context) error {
+	var b domain.InputUpdateBook
+	var id int
+	var err error
+
+	BookID := ctx.Param("book_id")
+	id, err = strconv.Atoi(BookID)
+	if err != nil {
+		return domain.ErrResponse(ctx, http.StatusBadRequest, errors.New("Invalid ID"))
+	}
+
+	bindErr := ctx.Bind(&b)
+	if bindErr != nil {
+		return domain.ErrResponse(ctx, http.StatusBadRequest, errors.New("data bind error"))
+	}
+
+	validationErr := config.Validator(b)
+	if validationErr != nil {
+		return domain.ErrValidResponse(ctx, http.StatusBadRequest, validationErr)
+	}
+
+	if uint(id) != b.BookID {
+		return domain.ErrResponse(ctx, http.StatusBadRequest, errors.New("Invalid ID"))
+	}
+
+	if b.Author != "" {
```

Comment:
ini bisa digabung jadi gini
```suggestion
if b.Author != "" && !config.ValidationBook(b.Author) {}
```